runtime/lua/vim/lsp/rpc.lua

```diff
@@ -149,7 +160,7 @@ M.client_errors = vim.tbl_add_reverse_lookup(M.client_errors)
 --- Constructs an error message from an LSP error object.
 ---
 ---@param err (table) The error object
----@returns (string) The formatted error message
+---@return (string) #The formatted error message
```

Comment:
```suggestion
---@return (string) The formatted error message
```

Reply:
The `#` is needed here (see [luals annotations documentation](https://github.com/LuaLS/lua-language-server/wiki/Annotations#return)) because return is expecting a name for the returned value and the `#` stops it from treating `The` as that name

Reply:
Then we can do something like `@return string The formatted error message`. Here `string` is a valid lua return type.

Reply:
The problem is not the type, but the fact that there is no name for the returned value. To omit the `#`, it should be something like `---@return string error_msg The error message returned`

Reply:
The `#` is fine; the parens are not (they are added by the `gen_vimdoc` script in the docs).

Reply:
@TheLeoP you are right, `#` is needed to document a "unnamed" return value. But the problem is that, although it's not the scope of this PR, neovim is not currently strictly following the syntax of lua-ls annotations; so the generated vimhelp doc contains weird string (it's a problem of `gen_vimdoc`). We would need a more systematic fix to make luadoc conforming to lua-ls annotations, but in my opinion it'd be good to make the generated documentation doesn't contain weird characters.